src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/FileResultExecutorBase.cs

```diff
@@ -60,7 +60,7 @@ internal enum PreconditionState
 
             var request = context.HttpContext.Request;
             var httpRequestHeaders = request.GetTypedHeaders();
-            var preconditionState = GetPreconditionState(httpRequestHeaders, lastModified, etag);
+            var preconditionState = GetPreconditionState(httpRequestHeaders, Logger, lastModified, etag);
```

Comment:
Why does this need to be passed around? Can the Logger property not be used directly?

Reply:
Because GetPreconditionState is a static method and I didn't want to modify it as we have tests for it.

Reply:
You have to modify it already to pass in the logger 😢. It would be better IMO to just make the code work the way we want rather than pass extra context around as a parameter